lib/mavenlink/model.rb

```diff
@@ -223,6 +223,28 @@ def association_load_filters
       {}
     end
 
+    def to_h
+      # Get a list of association names in this class
+      association_names = Mavenlink::Specificators::Association.apply(self.class)
+
+      pairs = association_names.collect do |association_name|
+        # Get the association object(s)
+        object_or_array = send(association_name)
```

Comment:
We should not be fetching every single association when `to_h`ing a model. This will end up making additional API calls to fetch all defined associations. Instead, we should look to what the model already has cached when `to_h`ing so that those additional objects can be added without any additional API calls.

For example, if a workspace is fetched without any additional `include`s then `to_h` should not return any additional associated objects because they are not present on the workspace. But if a workspace was loaded including `custom_field_values` and `primary_maven`, then only those should be added when `to_h`ing because the workspace record already has those cached and additional API calls are not required.

I also tried running this as-is on a workspace and it errored with ```NoMethodError (undefined method `attributes' for #<Mavenlink::User:0x00007fb16bbd4858>)```